src/actions.cpp

```diff
@@ -186,9 +186,10 @@ bool Actions::registerEvent(Event_ptr event, const pugi::xml_node& node)
 	return false;
 }
 
-bool Actions::registerLuaEvent(Event* event)
+bool Actions::registerLuaEvent(Action* event)
 {
-	Action_ptr action{ static_cast<Action*>(event) }; //event is guaranteed to be an Action
+	Event_ptr ptr = Event_ptr(event);
+	Action_ptr action{ static_cast<Action*>(ptr.release()) }; //event is guaranteed to be an Action
```

Comment:
@djarek just a quick question, wouldn't be `ptr.reset()` here better as it does not memory leak if we do not call delete on the raw pointer aswell? it maybe makes sense to use `ptr.release` if the event is guaranteed to be loaded again but on a reload it would memory leak if for example I removed that script from my files before reloading or not?

Reply:
`Action_ptr` is a `unique_ptr`, so there's no leak here. Also, if we did `reset()` here it would destroy the object and invalidate the pointer.

Reply:
You can drop `ptr` and assign the pointer directly into a `unique_ptr` (since the cast is no longer required):
```suggestion
	Action_ptr action{ event };
```

Reply:
ok that makes sense now, been reading a bit about unique pointers just now, was curious if `ptr.release` was called once `action` gives up ownership or if it's right away thanks for clearing that up

Reply:
ahh ok because I already push an `Action*` pointer into the function, makes sense actually.